### PR TITLE
add affinity config in fluentd deployment

### PR DESCRIPTION
**Pull request by vsinghal13**
###### Description
This PR exposes the use of `podAntiAffinity` to better schedule the fluentd replicas and prometheus pods across nodes.

By default, we are using a `preferredDuringSchedulingIgnoredDuringExecution` configuration alias `soft` which will check the conditions and schedule accordingly even if the conditions are not met. 

The `requiredDuringSchedulingIgnoredDuringExecution` configuration alias `hard` will keep the pods in Pending state if any of the conditions are not met.

Conditions:
- Don't schedule two fleuntd replicas on the same node
- Don't schedule prometheus pod along with fluentd 

###### Testing performed

- [x] ci/build.sh
- [x] Redeploy fluentd and fluentd-events pods
- [x] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/templates/deployment.yaml [hunk 1]
```diff
@@ -24,6 +24,43 @@ spec:
...
+          - labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - {{ template "sumologic.labels.app" . }}
+              - key: app
+                operator: In
+                values:
+                - prometheus-operator-prometheus
```
Comment (vsinghal13) [inline deploy/helm/sumologic/templates/deployment.yaml]: This section checks for the fluent app label.

File: deploy/helm/sumologic/templates/deployment.yaml [hunk 2]
```diff
@@ -24,6 +24,43 @@ spec:
...
+{{- end }}
+{{- if eq .Values.deployment.podAntiAffinity "hard"}}
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - {{ template "sumologic.labels.app" . }}
```
Comment (vsinghal13) [inline deploy/helm/sumologic/templates/deployment.yaml]: This section checks for the fleuntd app label.

File: deploy/helm/sumologic/templates/deployment.yaml [hunk 3]
```diff
@@ -24,6 +24,43 @@ spec:
...
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - {{ template "sumologic.labels.app" . }}
+                - key: app
+                  operator: In
+                  values:
+                  - prometheus-operator-prometheus
+              topologyKey: "kubernetes.io/hostname"
```
Comment (vsinghal13) [inline deploy/helm/sumologic/templates/deployment.yaml]: The `topologyKey` is the label for a particular node.

File: deploy/kubernetes/kube-prometheus-sumo-logic-mixin.libsonnet [hunk 4]
```diff
@@ -6,7 +6,7 @@
   sumologicCollector:: {
     remoteWriteConfigs+: [
       {
-        url: $._config.sumologicCollectorSvc + "prometheus.metrics.state",
```
Comment (perk-sumo) [inline deploy/kubernetes/kube-prometheus-sumo-logic-mixin.libsonnet]: Are those changes OK? Why they were expanded now with this PR?

Comment (vsinghal13) [reply deploy/kubernetes/kube-prometheus-sumo-logic-mixin.libsonnet]: This has been fixed, by merging master into the branch.

File: deploy/helm/sumologic/values.yaml [hunk 5]
```diff
@@ -8,6 +8,8 @@ nameOverride: ""
 deployment:
   nodeSelector: {}
   tolerations: {}
+  affinity: {}
+  podAntiAffinity: "soft"
```
Comment (samjsong) [inline deploy/helm/sumologic/values.yaml]: can you also leave a comment here for the customers that the accepted values are "soft" and "hard", and what those mean?

**Pull request merged**